src/DbBackupsTracker.cpp

```diff
@@ -0,0 +1,217 @@
+#include "DbBackupsTracker.h"
+
+#include <QDebug>
+#include <QException>
+#include <QFile>
+#include <QJsonDocument>
+#include <QJsonObject>
+#include <QSettings>
+
+DbBackupsTracker::DbBackupsTracker(QObject* parent)
+    : QObject(parent)
+{
+    loadTracks();
+
+    connect(&watcher, &QFileSystemWatcher::fileChanged,
+        this, &DbBackupsTracker::checkDbBackupSynchronization);
+}
+
+DbBackupsTracker::~DbBackupsTracker()
+{
+}
+
+QString
+DbBackupsTracker::getTrackPath(const QString& cpz) const
+{
+    return tracks.value(cpz, "");
+}
+
+QByteArray
+DbBackupsTracker::getCPZ() const
+{
+    return cpz;
+}
+
+int DbBackupsTracker::getCredentialsDbChangeNumber() const
+{
+    return credentialsDbChangeNumber;
+}
+
+QString DbBackupsTracker::readFile(QString path) const
+{
+    QString content;
+    QFile f(path);
+    if (f.exists()) {
+        f.open(QIODevice::ReadOnly);
+        content = f.readAll();
+    }
+    f.close();
+    return content;
+}
+
+int DbBackupsTracker::extractCredentialsDbChangeNumber(const QString& content) const
+{
+    int cn = 0;
+    QJsonDocument d = QJsonDocument::fromJson(content.toLocal8Bit());
+    if (d.isObject()) {
+        QJsonObject root = d.object();
+        if (root.contains("credentialsDbChangeNumber"))
```

Comment:
are you sure that works for both encrypted and non encrypted versions of the export file?

Reply:
Fixed in d81d76c